cpp/src/arrow/ipc/feather-test.cc

```diff
@@ -406,6 +406,89 @@ TEST_F(TestTableWriter, PrimitiveNullRoundTrip) {
   }
 }
 
+class TestTableWriterSlice : public TestTableWriter,
+                             public ::testing::WithParamInterface<std::tuple<int, int>> {
+};
+
+TEST_P(TestTableWriterSlice, SliceRoundTrip) {
+  auto p = GetParam();
+  auto start = std::get<0>(p);
+  auto size = std::get<1>(p);
+
+  std::shared_ptr<RecordBatch> batch;
+  ASSERT_OK(MakeIntBatchSized(start * 2, &batch));
+  batch = batch->Slice(start, size);
+
+  ASSERT_OK(writer_->Append("f0", *batch->column(0)));
```

Comment:
This part and below is still duplicated in the other two test cases. Do you think you can factor that out, for example as a template function?

Reply:
You are right the duplication can be removed without adding any complexity.